here-naksha-lib-jbon/src/jvmMain/kotlin/com/here/naksha/lib/jbon/JvmMapApi.kt

```diff
@@ -26,8 +31,14 @@ class JvmMapApi : IMapApi {
         m[key] = value
         return old
     }
+
     override fun remove(map: IMap, key: String): Any? = modify(map).remove(key)
     override fun clear(map: IMap) = modify(map).clear()
     override fun keys(map: IMap): Array<String> = read(map).keys.toTypedArray()
     override fun iterator(map: IMap): IMapIterator = JvmMapIterator(read(map))
+    override fun overrideBy(map1: IMap, map2: IMap): IMap {
+        val merged = newMap()
+        (map1 as JvmMap + map2 as JvmMap).iterator().forEach { merged.put(it.key, it.value) }
```

Comment:
This looks odd. What does the plus operator do here? If it adds the two maps, why to we then have to merge again? Can't we simply return (map1 as JvmMap + map2 as JvmMap)?

Reply:
'+' is kotlin operator and returns LinkedMap, which can't be cast to JvmMap, but yeah, you right, we don't have to iterate by hand, I'll change this to (putAll)